src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsResource.cs

```diff
@@ -47,8 +47,29 @@ public class AzureEventHubsResource(string name, Action<AzureResourceInfrastruct
     /// </summary>
     public ReferenceExpression ConnectionStringExpression =>
         IsEmulator
-        ? ReferenceExpression.Create($"Endpoint=sb://{EmulatorEndpoint.Property(EndpointProperty.Host)}:{EmulatorEndpoint.Property(EndpointProperty.Port)};SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true;")
-        : ReferenceExpression.Create($"{EventHubsEndpoint}");
+        ? ReferenceExpression.Create($"Endpoint=sb://{EmulatorEndpoint.Property(EndpointProperty.Host)}:{EmulatorEndpoint.Property(EndpointProperty.Port)};SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=SAS_KEY_VALUE;UseDevelopmentEmulator=true{BuildEntityPath()}")
+        : ReferenceExpression.Create($"{EventHubsEndpoint}{BuildEntityPath()}");
+
+    private string BuildEntityPath()
+    {
+        if (!Hubs.Any(hub => hub.IsDefaultEntity))
+        {
+            // Of zero or more hubs, none are flagged as default
+            return string.Empty;
+        }
+
+        try
+        {
+            // Of one or more hubs, only one may be flagged as default
+            var defaultEntity = Hubs.Single(hub => hub.IsDefaultEntity);
+
+            return $";EntityPath={defaultEntity.Name}";
+        }
+        catch (InvalidOperationException ex)
+        {
+            throw new InvalidOperationException("Only one EventHub can be configured as the default entity.", ex);
+        }
+    }
```

Comment:
1. Maybe a bit cleaner to use the ReferenceExpressionBuilder and a single method to build the connection string like https://github.com/dotnet/aspire/blob/3cf908b66566cfb8e3112545d931a24d5d6cfe15/src/Aspire.Hosting.MongoDB/MongoDBServerResource.cs#L56
2. Do we want this to throw here? Or somewhere else?

Reply:
1. Will take a look at this
2. That's my feeling too, but I'm not exactly sure where else and/or how to do that. Suggestions?

Reply:
Did you change something?

Reply:
btw, I changed the final exception to DistributedApplicationException as this seems to be the thing everywhere else for top level throws